packages/ui/src/components/va-tabs/hooks/useTab.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  computed,
+  inject,
+  onBeforeUnmount,
+  onMounted,
+  ref,
+  Ref,
+  SetupContext, unref,
+} from 'vue'
+import { ClassComputed, ComputedStyle, TabsServiceKey } from '../types'
+import useKeyboardOnlyFocus, { KeyboardFocusListeners } from '../../../composables/useKeyboardOnlyFocus'
+import { useColor } from '../../../composables/useColor'
+import { RouterLinkProps, useRouterLink } from '../../../composables/useRouterLink'
+import { TabsService } from './useTabsService'
+
+export interface TabProps {
+  color: string,
+  icon: string,
+  label: string,
+  disabled: boolean,
+  name: string | number | undefined,
+  tag: string,
+}
+
+export interface TabComponent {
```

Comment:
Do you really need all these properties in TabsService? Too much here, I think. Let's use in tab service only props, that we need.

Reply:
Any reason to use interface at all? Tabs and Tab components are heavily coupled, so there isn't much need to provide an interface, we can mention component type directly.

Reply:
Fixed to provide/inject methods.